src/wp-includes/blocks.php

```diff
@@ -1166,12 +1166,32 @@ function parse_blocks( $content ) {
  */
 function do_blocks( $content ) {
 	$blocks = parse_blocks( $content );
+
+	/**
+	 * Filter to allow plugins to inspect or change the array of parsed block
```

Comment:
a bit of a pedantic note, but this is a tree of parsed blocks. we are probably inconsistent in Core in our use of the terms, but it could help to subtly communicate that there's nesting and recursion involved - it's not a flat array

Reply:
Done in https://github.com/WordPress/wordpress-develop/pull/5023/commits/c5cfbc393562991f969e14d21ce9c76dc4a03f92.